src/img/node_modules/imgadm.js

```diff
@@ -314,7 +253,8 @@ var cacheUpdate = function (callback) {
 
         // SmartOS DNS is disabled by default
         dns.resolve4(options.hostname, function (err, addresses) {
-            var client;
+            var args = ['--insecure'];
```

Comment:
If nothing else, the existing code will validate certificates by default and I think it's entirely inappropriate to disregard certificate errors.